dll/cpl/sysdm/smbios.c

```diff
@@ -511,15 +662,33 @@ BOOL GetSystemName(PWSTR pBuf, SIZE_T cchBuf)
     }
 
     // workaround for LENOVO notebooks
-    if (!wcscmp(ven, L"LENOVO"))
+    if (!wcsicmp(ven, L"LENOVO"))
     {
         StringCchCopyW(ven, _countof(ven), L"Lenovo");
 
         if (DmiStrings[SYS_VERSION] != NULL &&
-            stricmp(DmiStrings[SYS_VERSION], "Lenovo") &&
-            stricmp(DmiStrings[SYS_VERSION], "Lenovo Product") &&
+            !strncmp(DmiStrings[SYS_VERSION], "ThinkPad   ", 11))
+        {
+            DmiStrings[SYS_VERSION][8] = L'\0';
+        }
+        if (wcslen(dev) > 0 &&
+            DmiStrings[SYS_VERSION] != NULL &&
+            (!strcmp(DmiStrings[SYS_VERSION], "IdeaCentre") ||
+            !strcmp(DmiStrings[SYS_VERSION], "ThinkPad")))
```

Comment:
No `IdeaPad`?

Reply:
ThonkPad 🤔

Reply:
In some Lenovo dumps the SYS_VERSION string contains device family instead of version, this code portion handles these cases.

I have a bunch of IdeaPad dumps, but they don't contain this bug. :slightly_smiling_face: